### DISCUSSION TITLE
tlg-catalog/stable/tables/efficacy/rspt01

### DISCUSSION THREAD

**Discussion by giscus** [insightsengineering/tlg-catalog#43] [General]
# tlg-catalog/stable/tables/efficacy/rspt01

Best Overall Response

https://insightsengineering.github.io/tlg-catalog/stable/tables/efficacy/rspt01.html

**Comment by Teninq** [comment]
`estimate_odds_ratio(
    vars = "is_rsp",
    variables = list(strata = strata, arm = "ARM"),
    table_names = "est_or_strat"
  )`
In this function, do we need to specifiy arm? Or this function can directly get arm information from `lyt`. I noticed that the arm here is by default set to "NULL" when we perform the unstratified analysis, which is weird to me. Could you explain how this function get the `arm` information for unstratified analysis?

**Reply by shajoezhu** [reply]
for stratified analysis, please refer to the s_ function in tern, 

s_odds_ratio(): Statistics function which estimates the odds ratio between a treatment and a control. Note that a variables list with arm and strata names needs to be passed if a stratified analysis is required.

**Comment by Teninq** [comment]
The level of response seems not in a good sequence, the `Not Evaluable (NE)` should be displayed after `Stable Disease (SD) `.

I have updated the `d_onco_rsp_label` function and you can take it as reference.
```d_onco_rsp_label1 <- function (x)
{
  x <- as.character(x)
  desc <- c(CR = "Complete Response (CR)", PR = "Partial Response (PR)",
            MR = "Minimal/Minor Response (MR)", MRD = "Minimal Residual Disease (MRD)",
            SD = "Stable Disease (SD)", PD = "Progressive Disease (PD)",
            `NON CR/PD` = "Non-CR or Non-PD (NON CR/PD)",  NE = "Not Evaluable (NE)",
            `NE/Missing` = "Missing or unevaluable", Missing = "Missing",
            `NA` = "Not Applicable (NA)", ND = "Not Done (ND)")
  values_label <- vapply(X = x, FUN.VALUE = character(1), function(val) {
    if (val %in% names(desc))
      desc[val]
    else val
  })
  return(factor(values_label, levels = desc))
}
```

**Reply by Melkiades** [reply]
Open an issue regarding this and someone (maybe me) will work it in

**Comment by staceyfu1029** [comment]
Is there a way to add blank rows between layers to improve the readability?

**Reply by Melkiades** [reply]
If I understood correctly you need to use the parameter `section_div` from `rtables::split_rows_by()`
